service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/CreateKubernetesNamespaceStep.java

```diff
@@ -43,7 +45,12 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     var clusterResource =
         kubernetesClientProvider
             .getClusterResource(workspaceId)
-            .orElseThrow(() -> new RuntimeException("No shared cluster found"));
+            .orElseThrow(
+                () ->
+                    new MissingResourceException(
```

Comment:
Should the tests then be updated to handle the more explicit exception?

Reply:
The existing tests for this just care about success vs failure.

There isn't anything that actually cares about the type of exception, which makes sense because an exception thrown in the `doStep` is handled directly by stairway anyway. This will mostly just improve error messages to be more clear.